File: hencoderl28_annotation_processor/src/main/java/com/example/hencoderl28_annotation_processor/MainActivity.java
package com.example.hencoderl28_annotation_processor;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import com.example.hencoderl28_lib.Binding;
import com.example.hencoderl28_lib_annotation.BindView;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.text)
    TextView mTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Binding.bind(this);

        mTextView.setText("通过annotationProcessor进行绑定");

        mTextView.setOnClickListener(view -> {
            Intent intent = new Intent(this, SecondActivity.class);
            startActivity(intent);
        });
    }
}

File: coordinaotrlayoutdemo/src/main/java/com/example/coordinaotrlayoutdemo/MainActivity.kt
package com.example.coordinaotrlayoutdemo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.btn_with_toolbar).setOnClickListener( View.OnClickListener {
            startActivity(Intent(this, ToolbarWithCoordinatorLayoutActivity::class.java))
        })
        findViewById<View>(R.id.btn_with_toolbar_snap).setOnClickListener( View.OnClickListener {
            startActivity(Intent(this, ToolbarSnapActivity::class.java))
        })
        findViewById<View>(R.id.btn_with_view_pager).setOnClickListener( View.OnClickListener {
            startActivity(Intent(this, CoordinatorLayoutAndViewPagerActivity::class.java))
        })
        findViewById<Button>(R.id.btn_custom_item_decoration).setOnClickListener {
            ScrollToPosActivity.start(this);
         }

        findViewById<Button>(R.id.btn_custom_simple_behavior).setOnClickListener {
            startActivity(Intent(this, SimpleBehaviorActivity::class.java))
        }

        findViewById<Button>(R.id.btn_nestscroll).setOnClickListener {
            startActivity(Intent(this, NestedScrollViewActivity::class.java))
        }
        findViewById<Button>(R.id.btn_space).setOnClickListener {
            startActivity(Intent(this, SpaceRvActivity::class.java))
        }
    }
}
